src/sentry/utils/snuba.py

```diff
@@ -922,6 +924,30 @@ def __init__(
         self.is_grouprelease = is_grouprelease
         self.kwargs = kwargs
 
+        # Groups can be merged together, but snuba is immutable(ish). In order to
+        # account for merges, here we expand queries to include all group IDs that have
+        # been merged together.
+
+        if options.get("snuba.preprocess-group-redirects"):
+            if "group_id" in self.filter_keys:
+                self.filter_keys["group_id"] = get_all_merged_group_ids(
```

Comment:
Won't this only do 1 level of expansion in the merge graph? Is this intentional?

e.g. if Group A was merged into Group B and Group B was merged into Group C, we have:
```
GroupRedirect(group_id=B, previous_group_id=A)
GroupRedirect(group_id=C, previous_group_id=B)
```

And then if I group ID C as a filter key, the set of merged group IDs returned from `get_all_merged_group_ids` would be {B, C}.